### PR TITLE
Update Google SignIn API

### LINKED ISSUE #277: Remove deprecated `GoogleSignIn` api
In the current state, the application uses a deprecated API to Sign In via Google.
The ideal app should use the newer Google API.

### PR DESCRIPTION
**Pull request by matttsss**
## What is this PR?

This PR aims to remove the deprecated version of the Google SignIn API that we used. It now uses the new API that does not launch any Intents.

## Future work (#274):
With this new API, the `Authenticator` system is not necesary since we don't need compose functions to authenticate.
My next PR will be aimmed at integrating authentication code within the `UserViewModel`.

### Test Coverage:
image

Closes #277

### CODE REVIEW

**Review by sean-prz — APPROVED** [review]
Everything still works, I tested with my account/new account and as a guest. 

For the code I can't provide any feedback, it's just too in depth on the topic of authentification. It's just useless for me to dive as deep as you on this.

It is a bit hard to follow the code.
When I read it, I have to jump from AuthenticatorImpl.kt  to Authenticator.kt  to check the default button, then back to AuthenticatorImpl to see the two params passed  then further down the code to see the actual params, which themselves are defined in another function.
Would be nice if you could write a summary or draw a graph so we get a grasp of what's happening. 

I will approve regardless because I don't think any one but the reviewer will have to understand this code

**Comment by matttsss** [comment]
@sean-prz Yes the process is a bit complicated... I am currently working on simplifying it by integrating the `Authenticator` to the `UserViewModel` to get one API

**Review by Armetiis — APPROVED** [review]
To be completely honest I have nothing to say. Works good. I'd like to comment on the code but I must admit that it's a bit too niche and as Sean implied, the research necessary to understand this code is quite tedious to be done by all the reviewers in addition to you. 

I still appreciate that you left good documentation and comments to help potential future people maintain the code
